python/lsst/sims/photUtils/SignalToNoise.py

```diff
@@ -0,0 +1,552 @@
+import numpy
+from .Sed import Sed
+from .Bandpass import Bandpass
+from lsst.sims.photUtils import LSSTdefaults
+
+__all__ = ["calcNeff", "calcInstrNoiseSq", "calcTotalNonSourceNoiseSq", "calcSNR_sed",
+          "calcM5", "calcSkyCountsPerPixelForM5", "calcSkyCountsForM5", "calcGamma", "calcSNR_m5",
+          "calcAstrometricError", "magErrorFromSNR", "calcMagError_m5", "calcMagError_sed"]
+
+
+def calcNeff(seeing, platescale):
+    """
+    Calculate the effective number of pixels in a double Gaussian PSF
+
+    @param [in] seeing in arcseconds
+
+    @param [in] platescale in arcseconds per pixel
+
+    @param [out] the effective number of pixels contained by a double
+    Gaussian PSF
+
+    see equation 31 of the SNR document
+    https://docushare.lsstcorp.org/docushare/dsweb/ImageStoreViewer/LSE-40
+    """
+    return 2.436*(seeing/platescale)**2
+
+
+def calcInstrNoiseSq(photParams):
+    """
+    Combine all of the noise due to intrumentation into one value
+
+    @param [in] photParams is an instantiation of the
+    PhotometricParameters class that carries details about the
+    photometric response of the telescope.
+
+    @param [out] The noise due to all of these sources added in quadrature
+    in ADU counts
+    """
+
+    return (photParams.nexp*photParams.readnoise**2 + \
+           photParams.darkcurrent*photParams.exptime*photParams.nexp + \
+           photParams.nexp*photParams.othernoise**2)/(photParams.gain*photParams.gain)
+
```

Comment:
It's probably more efficient to just 'return' the value, but it might be more readable to refactor this a little .. have a line which calculates the value (into a variable) and then returns that variable, mostly because of the length of the value you're calculating. 

# Inst noise in electrons
instNoiseSq = photParams.nexp * ( photParams.readnoise**2 + photParams.darkcurrent*photParams.exptime + photParamsothernoise**2)   
# Convert to counts
instNoiseSq = instNoiseSq / photParams.gain**2
return instNoiseSq